src/transforms/RemoveFunctions.cpp

```diff
@@ -0,0 +1,115 @@
+
+namespace previrt {
+namespace transforms {
+
+using namespace llvm;
+using namespace std;
+
+struct RemoveFunctions : public FunctionPass {
+	
+	static char ID;
+	RemoveFunctions() : FunctionPass(ID) {}
+        
+	bool runOnFunction(Function &F) override {
+
+		Module* module = F.getParent();
+		vector<string> removeFunctions;
+		stringstream string_to_stream(function_list);
+
+		while (string_to_stream.good()) {
+			string substr;
+        		getline(string_to_stream, substr, ',');
+        		removeFunctions.push_back(substr);
+    		}
+
+//		vector<string> removeFunctions{"func1","func2","func3","func4","add"};
+    
+		// Declare printf function
+//		Type *intType = Type::getInt32Ty(module->getContext());
+//		std::vector<Type *> printfArgsTypes({Type::getInt8PtrTy(module->getContext())});
+//		FunctionType *printfType = FunctionType::get(intType, printfArgsTypes, true);
+//		auto *printfFunc = dyn_cast<Function>(module->getOrInsertFunction("printf", printfType).getCallee()); 
+		Type* charType = Type::getInt8PtrTy(module->getContext());
+                FunctionType *printf_type = FunctionType::get(charType,true);
+                auto *printfFunc = cast<Function>(module->getOrInsertFunction("printf", printf_type).getCallee());
+
+		assert(printfFunc && "printf not found in module");
+
+		// Declare exit function
+		Type *voidType = Type::getVoidTy(module->getContext());
+		vector<Type *> exitArgsTypes;
+		exitArgsTypes.push_back(Type::getInt32Ty(module->getContext()));
+		FunctionType *exitType = FunctionType::get(voidType, exitArgsTypes, false);
+		Function *exitFunc = dyn_cast<Function>(module->getOrInsertFunction("exit", exitType).getCallee());
+
+		string functionName = F.getName().str();
+		errs()<<"\nRemoveFunctions pass: Function being analysed : "<<functionName<<"\n";
+	
+	       	if( find(removeFunctions.begin(), removeFunctions.end(), functionName) != removeFunctions.end()  ){
+			errs()<<"\nRemoveFunctions: User asked this function("<<functionName<<") to be removed\n";
+	
+			BasicBlock * entryBB = &F.getEntryBlock();
+			BasicBlock *emptyBB = BasicBlock::Create(module->getContext(), "empty");
+        	    	emptyBB->insertInto(&F,entryBB);
+			IRBuilder<> Builder(emptyBB);
+			//Add printf call in empty BB
+            		Value *str = Builder.CreateGlobalStringPtr("You asked to remove this function. Program terminating!\n", "str");
+            		std::vector<Value *> removeMessage({str});
+            		Builder.CreateCall(printfFunc, removeMessage, "unreahcable");
+           		//Add exit call in empty BB
+            		Value *status = ConstantInt::get(Type::getInt32Ty(module->getContext()),(-10));
+            		Builder.CreateCall(exitFunc, status, "");
+            		Type* returnType = F.getReturnType();
+            		//Add terminator instruction
+            		ReturnInst::Create(module->getContext(), UndefValue::get(returnType), emptyBB);
```

Comment:
Add a LLVM `unreachable` instruction. Return something with undef value is not a good idea.

Reply:
Done. I